src/connection_manager.c

```diff
@@ -98,9 +98,13 @@ static void qd_server_config_free(qd_server_config_t *cf)
     if (!cf) return;
     free(cf->host);
     free(cf->port);
-    free(cf->name);
     free(cf->role);
-    free(cf->sasl_mechanisms);
+    if (cf->name)            free(cf->name);
```

Comment:
This is a good change. It makes sure that it frees all the components of qd_server_config_t